requirements/pre-dev.txt

```diff
@@ -1 +1 @@
-pylint>=2.13.0,<3
+ruff
```

Comment:
Ruff is not very stable and keeps adding rules. We should try to ping the version otherwise once it adds a new rule the build might fail

Reply:
Any build failures you have experienced? I'm gonna pin to 0.0.247 in that case.

Reply:
they added `too-many-arguments` in the past few weeks and when we upgrade there were many violations in sam-t codebase